src/Multicall3Delegatecall.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+/// @title Multicall3Delegatecall
+/// @notice Aggregate delegate calls, ensuring each returns success if required.
+///         This contract is intended to faciliate the use of delegatecalls from a Safe, while
+///         also enabling us to continue using the base-contracts library which we depend on to
+///         facilitate simulating and signing multisig transactions.
+///         This contract is based on Multicall3 deployed on many evm chains
+///         at 0xcA11bde05977b3631167028862bE2a173976CA11. It has been modified to
+///         1. Perform delegatecalls instead of calls.
+///         2. Utility getter functions have been removed.
```

Comment:
```suggestion
///         at 0xcA11bde05977b3631167028862bE2a173976CA11. It has been modified to
///         1. Perform delegatecalls instead of calls.
///         2. Remove utility getter functions that are unneeded for this use case.
```